### PR TITLE
Update documentation on user storage provider in Quarkus

### LINKED ISSUE #17394: Update documentation on user storage provider in Quarkus
Update documentation in https://github.com/keycloak/keycloak-documentation/tree/release/21.0/server_development/topics/user-storage to match Quarkus development for legacy store.

Determine correct target guide, either in https://github.com/keycloak/keycloak-documentation/tree/main/server_development or https://github.com/keycloak/keycloak/tree/main/docs/guides

### PR DESCRIPTION
**Pull request by martin-kanis**
Closes #17394

### CODE REVIEW

**Review by vramik — DISMISSED** [review]
Thank you @martin-kanis!

File: docs/documentation/server_development/topics/user-storage/configuration.adoc [hunk 1]
```diff
@@ -76,7 +76,9 @@ The next thing we want to do is to verify that this file exists on disk. We do n
     }
 ----
 
-In the `validateConfiguration()` method we get the configuration variable from the `ComponentModel` and we check to see if that file exists on disk. Notice that we use the `org.keycloak.common.util.EnvUtil.replace()` method. With this method any string that has `${}` within it will replace that with a system property value. The `${jboss.server.config.dir}` string corresponds to the `configuration/` directory of our server and is really useful for this example.
+In the `validateConfiguration()` method we get the configuration variable from the `ComponentModel` and we check to see if that file exists on disk.
```
Comment (andymunro) [inline docs/documentation/server_development/topics/user-storage/configuration.adoc:79]:
```suggestion
The `validateConfiguration()` method provides the configuration variable from the `ComponentModel` to verify if that file exists on disk.
```

File: docs/documentation/server_development/topics/user-storage/configuration.adoc [hunk 2]
```diff
@@ -76,7 +76,9 @@ The next thing we want to do is to verify that this file exists on disk. We do n
...
+Notice that we use the `org.keycloak.common.util.EnvUtil.replace()` method. With this method any string that has `${}` within it will replace that with a system property value.
+The `${jboss.server.config.dir}` string corresponds to the `conf/` directory of our server and is really useful for this example.
```
Comment (andymunro) [inline docs/documentation/server_development/topics/user-storage/configuration.adoc:81]:
```suggestion
Notice that the use of the `org.keycloak.common.util.EnvUtil.replace()` method. With this method any string that includes`${}` will replace that value with a system property value.
The `${jboss.server.config.dir}` string corresponds to the `conf/` directory of our server and is really useful for this example.
```

File: docs/documentation/server_development/topics/user-storage/registration-query.adoc [hunk 3]
```diff
@@ -136,95 +136,55 @@ by our example provider. Let's look at implementing this interface.
...
+                .skip(firstResult)
+                .map(username -> getUserByUsername(realm, username))
+                .limit(maxResults);
     }
 ----
 
-The `getUsers()` method iterates over the key set of the property file, delegating to `getUserByUsername()` to load a user.
+The first declaration of `searchForUserStream()` takes a `String` parameter. In this example, it represents a username that you want to search by. This string can be a substring, which is why we use the `String.contains()`
+method when doing our search. Notice that we use `*` to indicate we want to list all users.
+The method iterates over the key set of the property file, delegating to `getUserByUsername()` to load a user.
```
Comment (andymunro) [inline docs/documentation/server_development/topics/user-storage/registration-query.adoc:152]:
```suggestion
The first declaration of `searchForUserStream()` takes a `String` parameter. In this example, the parameter represents a username that you want to search by. This string can be a substring, which explains the choice of the `String.contains()`
method when doing the search. Notice the use of `*` to indicate to request a list all users.
The method iterates over the key set of the property file, delegating to `getUserByUsername()` to load a user.
```

File: docs/documentation/server_development/topics/user-storage/registration-query.adoc [hunk 4]
```diff
@@ -136,95 +136,55 @@ by our example provider. Let's look at implementing this interface.
...
+        // if we are not searching by username, return all users
+        return searchForUserStream(realm, "*", firstResult, maxResults);
     }
 ----
 
-The `searchForUser()` method that takes a `Map` parameter can search for a user based on first, last name, username, and email.
-We only store usernames, so we only search based on usernames. We delegate to `searchForUser()` for this.
+The `searchForUserStream()` method that takes a `Map` parameter can search for a user based on first, last name, username, and email.
+We only store usernames, so we only search based on usernames. Only exception is when `Map` parameter doesn't contain `username` attribute
+in which case we return all users. We delegate to `searchForUserStream(realm, search, firstResult, maxResults)` for this.
```
Comment (andymunro) [inline docs/documentation/server_development/topics/user-storage/registration-query.adoc:172]:
```suggestion
Only usernames are stored, so the search is based only on usernames except when the `Map` parameter does not contain the `username` attribute. In this case, all users are returned. In that situation,  use `searchForUserStream(realm, search, firstResult, maxResults)` .
```

File: docs/documentation/server_development/topics/user-storage/registration-query.adoc [hunk 5]
```diff
@@ -136,95 +136,55 @@ by our example provider. Let's look at implementing this interface.
...
     @Override
-    public List<UserModel> searchForUserByUserAttribute(String attrName, String attrValue, RealmModel realm) {
-        return Collections.EMPTY_LIST;
+    public Stream<UserModel> searchForUserByUserAttributeStream(RealmModel realm, String attrName, String attrValue) {
+        return Stream.empty();
     }
 ----
 
-We do not store groups or attributes, so the other methods return an empty list.
+We do not store groups or attributes, so the other methods return an empty stream.
```
Comment (andymunro) [inline docs/documentation/server_development/topics/user-storage/registration-query.adoc:189]:
```suggestion
Groups or attributes are not stored, so the other methods return an empty stream.
```

**Review by andymunro — CHANGES_REQUESTED** [review]
@mhajas Just a few suggestions. Thanks.

**Comment by martin-kanis** [comment]
I updated the PR according Andy's suggestions. @vramik @andymunro  Can you please re-review?

**Review by andymunro — APPROVED** [review]

**Review by vramik — APPROVED** [review]

**Review by mhajas — APPROVED** [review]
Approving per @vramik 's and @andymunro 's approvals

**Pull request merged**